src/Aspire.Dashboard/Authentication/ApiKeyAuthorizationHandler.cs

```diff
@@ -0,0 +1,31 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Authorization;
+
+namespace Aspire.Dashboard.Authentication;
+
+public class ApiKeyAuthorizationHandler : AuthorizationHandler<ApiKeyAuthorizationRequirement>
```

Comment:
In ASP.NET Core auth parlance, the Authentication handler deals with taking data from the HttpContext and turning it into an [AuthenticationTicket](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.authentication.authenticationticket?view=aspnetcore-8.0) (ClaimsPrincipal + AuthenticationProperties) and then AuhorizationHandlers should use that information to decide if the user is authorized.

So this should be implemented as an AuthenticationHandler a simple claims check in the `ApiKeyAuthorizationHandler`.

Reply:
Done.